### PR TITLE
Add host option

### LINKED ISSUE #232: github command set does not support enterprise api hosts
The command set for github assumes github.com (api.github.com). The code needs to be parameterized to allow github hosted elsewhere.

### PR DESCRIPTION
**Pull request by bhageena**

### CODE REVIEW

File: github/README.md [hunk 1]
```diff
@@ -31,14 +31,22 @@ Please make sure Commander is installed in your [Slack workspace](https://slack.
...
 
 ![Secret Creator](https://raw.githubusercontent.com/nimbella/command-sets/master/github/screenshots/secret_creator.png)
 
 For commands that only work with single repository the first repository in `github_repos` is used.
 
+### Github Enterprise
+
+Only for Github Enterprise version, you need to specify the host. Below are the ways to do it, you can follow any of them
+- Specify `--host` or `-h` while invoking the command.
+- Append `github_token` secret value with host name, followed by `@`, i.e. <your_access_token>@<your_github_host>.
```
Comment (rabbah) [inline github/README.md]:
```suggestion
- Append `github_token` secret value with host name, followed by `@`, i.e. <your_access_token>@<your_github_host>. For example `mysecret@example.com`.

```

**Review by rabbah — COMMENTED** [review]
For a subsequent iteration, I suggestion the following enhancements:
1. refactor common code across all the command sets into a utility class
2. add unit tests, lots of tests

File: github/README.md [hunk 2]
```diff
@@ -31,20 +31,27 @@ Please make sure Commander is installed in your [Slack workspace](https://slack.
...
 
 ![Secret Creator](https://raw.githubusercontent.com/nimbella/command-sets/master/github/screenshots/secret_creator.png)
 
 For commands that only work with single repository the first repository in `github_repos` is used.
 
+### Github Enterprise
+
+Only for Github Enterprise version, you need to specify the host. Below are the ways to do it, you can follow any of them
+- Specify `--host` or `-h` while invoking the command.
+- Append `github_token` secret value with host name, followed by `@`, i.e. <your_access_token>@<your_github_host>. For example `mysecret@example.com`.
```
Comment (rabbah) [inline github/README.md]:
```suggestion
- Append the host name to your `github_token` following this format: `access_token@github_host`. For example `mysecret@example.com`.

```

File: github/README.md [hunk 3]
```diff
@@ -31,20 +31,27 @@ Please make sure Commander is installed in your [Slack workspace](https://slack.
...
+- Create secret with key name `github_host` using `/nc secret_create`, add host as value.
```
Comment (rabbah) [inline github/README.md]:
```suggestion
- Create a secret with key name `github_host` using `/nc secret_create`, and add the host as the value.

```

File: github/README.md [hunk 4]
```diff
@@ -31,20 +31,27 @@ Please make sure Commander is installed in your [Slack workspace](https://slack.
...
+
+If you follow multiple ways, priority is decided top-down.
```
Comment (rabbah) [inline github/README.md]:
```suggestion
If you use more than one of the approaches listed above, the precedence is decided top-down (e.g., command line supersedes all others).
```

File: github/README.md [hunk 5]
```diff
@@ -82,14 +89,6 @@ Create an issue.
 
 ![GitHub create_issue command](https://raw.githubusercontent.com/nimbella/command-sets/master/github/screenshots/create_issue.png)
 
-### [`github_find_pr`](https://github.com/nimbella/command-sets/blob/master/github/packages/github/github_find_pr.js)
```
Comment (rabbah) [inline github/README.md:85]: why is this removed?

File: github/commands.yaml [hunk 6]
```diff
@@ -7,6 +7,8 @@ commands:
     options:
```
Comment (rabbah) [inline github/commands.yaml:7]: future idea: we should consider generating the commands configuration from a more concise grammar (like the syntax for the command) - this file is long and prone to error, and the semantic information can be encoded better (and more human friendly, eg: `command-name [-h host] [-r repo] ...`

**Review by rabbah — APPROVED** [review]
It's difficult to review a PR this big. A lot of code is amenable to unit tests as well. I will merge this PR but as follow-on work, we need to refactor the common code, and add unit tests.

File: github/README.md (hunk 5 above)
Comment (bhageena) [reply github/README.md:85]: It had similar functionality as [github_view_pr.js](https://github.com/nimbella/command-sets/blob/master/github/packages/github/github_view_pr.js) and the actual implementation file (github_find_pr.js) was removed long back, only the document was not updated.
